documentation/design-docs/data-catalog.md

```diff
@@ -339,6 +354,79 @@ and validating our data using dbt:
+  in SQL and moved into the dbt DAG. During the transition period, while some
+  transformations are still written in R, we will treat their output as if it
+  were raw data and reference it using `source()`. Any transformations that
+  can't be easily rewritten in SQL will continue to be defined this way in the
+  long term.
+* Intermediate or final transformations that require CPU- or memory-intensive
+  operations like running machine learning models will be defined in Python,
+  run as AWS Glue jobs, and defined as [ephemeral
+  models](https://docs.getdbt.com/docs/deploy/source-freshness) in the dbt DAG.
+  This will be true even in cases where the Glue jobs depend on models produced
+  by the dbt DAG, e.g. the tables produced by
+  [`model-sales-val`](https://github.com/ccao-data/model-sales-val). A bullet
+  below will explain how we will manage circular dependencies between these
+  services.
+* Glue jobs will be kept under version control and deployed to AWS using
+  [Terraform](https://www.terraform.io/) run in GitHub Actions on
+  commits to their repo's main branch. We will write a reusable [composite
+  action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action)
+  that performs the following operations:
+    1. Runs `terraform apply` to recreate the Glue job definition in AWS
+      1. In doing so, inserts the current Git SHA as a command argument in the
+         Glue job definition so that the job script can read the SHA and use it
+         for versioning.
+      2. Supports creating staging jobs that we can use for testing during CI.
+    2. Uploads the newest version of the script to the proper bucket in S3
+* There will be three expected ways in which we handle dependencies between
+  dbt and glue, depending on the direction of the dependency graph:
+    * In cases where dbt depends on the output of a Glue job (Glue -> dbt), we
+      will treat the Glue job output as an  or
+      a [source](https://docs.getdbt.com/docs/build/sources) in the DAG and
+      schedule the job as necessary to maintain freshness.
+        * If we would like to rebuild the dbt models every time the Glue
+          source data updates, we can schedule the job via GitHub Actions
+          instead of the Glue job scheduler and configure GitHub Actions to
+          rerun dbt in case of a successful Glue job run.
+    * In cases where a Glue job depends on the output of dbt (dbt -> Glue),
+      we will document the Glue job as an
+      [exposure](https://docs.getdbt.com/docs/build/exposures) in the DAG.
+      If we would like to ensure that we run the Glue job every time the
+      dbt source data updates, we can write a wrapper script around `dbt run`
+      that uses the Glue `StartJobRun` API
+      ([docs](https://docs.aws.amazon.com/glue/latest/webapi/API_StartJobRun.html))
+      to trigger a job run once the dbt build completes successfully.
```

Comment:
**suggestion:** I don't love this. Maybe we can instead integrate it with Actions and query the dbt state to trigger the job? Unsure what's possible here or how often this will really come up.

Reply:
I think that makes sense! It seems like it just depends on whether we want dbt to be coupled to the Glue job, or the Glue job to be coupled to dbt; given the dependency order here I agree that it probably makes sense to couple Glue to dbt, since dbt doesn't need to care about how its downstream consumers consume it. I adjusted this note in da2d0acf35495fff221b5c8675eb5ab398b1f429, but I'm up for tweaking it further if we come up with a better solution.